src/main/groovy/com/modelsolv/reprezen/soapui/RepreZenExporter.groovy

```diff
@@ -94,22 +94,51 @@ class RepreZenExporter {
 
 	private def exportRestService(RestService restService, ResourceAPI resourceAPI, DataModel dataModel) {
 		restService.resourceList.each {
-			def ServiceDataResource res = createResourceDefinition( it, resourceAPI, dataModel)
+			def ServiceDataResource res = toResourceDefinition( it, resourceAPI, dataModel)
 			resourceAPI.ownedResourceDefinitions.add(res)
 		}
 	}
 
-	private def ServiceDataResource createResourceDefinition( RestResource resource, ResourceAPI resourceAPI, DataModel dataModel ) {
-		ServiceDataResource result = restapiFactory.createObjectResource()
+	private def ServiceDataResource toResourceDefinition( RestResource resource, ResourceAPI resourceAPI, DataModel dataModel ) {
+		ServiceDataResource result = createResource(resource)
+
+		result.URI = createUriWithParameters(resource)
+
+		resource.restMethodList.each {
+			Method method = createMethod( it, resourceAPI )
+			result.methods.add(method)
+		}
+
+		resource.childResourceList.each {
+			ResourceDefinition res = toResourceDefinition( it, resourceAPI, dataModel)
+			resourceAPI.ownedResourceDefinitions.add(res )
+		}
+		Structure structure = datatypesFactory.createStructure()
+		structure.setName(result.getName() + "Type")
+		dataModel.getOwnedDataTypes().add(structure)
+		result.setType(structure)
+		return result
+	}
+
+	private ServiceDataResource createResource(RestResource resource) {
+		def result = isCollectionResource(resource) ? restapiFactory.createCollectionResource() : restapiFactory.createObjectResource();
 		result.name = normalize(resource.name)
-		if( hasContent(resource.description))
+		if (hasContent(resource.description)) {
 			addDocumentation(result, resource.description)
+		}
+		result
+	}
 
+	private boolean isCollectionResource(RestResource resource) {
+		resource.name.toLowerCase().contains("collection")
+	}
```

Comment:
Is this really the best way to determine whether the resource is a collection or object? E.g. if I were writing an API dealing with stamp collections, would I be screwed if I wanted to have an object resource named "StampCollection"? It would surprise me to learn that the distinction isn't built into the model, but I'm way less familiar with that than  you are.